improver/generate_ancillaries/generate_derived_solar_fields.py

```diff
@@ -127,6 +137,159 @@ def _initialise_input_cubes(
 
         return surface_altitude, linke_turbidity
 
+    def _get_irradiance_times(
```

Comment:
The name of this function "get" sounds like the times are being retrieved from somewhere. Suggest renaming, maybe to "calculate" or "construct". Alternatively, drop the prefix and shorten to "_irradiance_times".

Reply:
Good idea. I think I'll drop the prefix and use `_irradiance_times`.